### PR TITLE
Add guidelines

### PR DESCRIPTION
**Pull request by vladislavvalt**

### CODE REVIEW

File: GUIDELINE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Guideline
+
+These are some tips to help you pass this assignment.
+We want you to succeed.
+In case you don't manage to pass the review, we want you to avoid unnecessary effort since we respect both your effort and your time.
+
+* Provide unit and integration tests: **the solution that does not have tests rarely passes our review**
```
Comment (islammohamed) [inline GUIDELINE.md:7]: QQ: do you think if you make it optional, it shows by itself, how the engineer prioritization when it comes to testing? I assume if you insist on it, some will fake it even if it's not part of what they're developing normally.

File: GUIDELINE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+* Please submit a clean solution
```
Comment (islammohamed) [inline GUIDELINE.md:8]: the same here as above.

File: GUIDELINE.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    * No commented code
+    * Code must compile
+    * Tests are executable
+    * Code should be easy to read
+    * Better deliver less, but something that works
+* Find a way to showcase your skill in scope of the task
+  * If you are applying for Senior or Lead position, we expect you to manifest your skills, not merely meet the requirements
+* Do not overcomplicate
+    * We do not count lines of code, we evaluate the quality of the solution, the smaller it is, the better (yet it has to satisfy task requirements and showcase your skill - it's your task to strike a good balance between meeting requirements and brevity)
+    * We value your time and ours. We expect a single Controller, Service, and Repository. Feel free to create as many auxiliary classes as needed.
```
Comment (islammohamed) [inline GUIDELINE.md:18]: Here I see also a great direction and hint, that somebody can be used to show extra maybe fake effort, even if it's not his/her norm. wdyt?

**Comment by islammohamed** [comment]
I really like the initiative and great effort showing how we respect the time of the candidates as ours.
The main issue for me, this is part of a great experience which is being part of the daily routine we're doing, I'm afraid if we can give those fine-grained tricks, the developers may find their way presenting some out of their norm

File: GUIDELINE.md (hunk 1 above)
Comment (vladislavvalt) [reply GUIDELINE.md:7]: Yes, I agree with you, I guess I should remove the direct statement in bold and only give a hint.

Something like: "Please make sure that the reviewer can conveniently verify that the application works as expected".

**Comment by vladislavvalt** [comment]
I agree. We should be careful not to make it too easy to fake expertise and experience.
My main goal is to clarify some expectations from the candidates:
* We do not review to see huge applications
* We want applications to be clean (I guess I should not provide too many hints here)

File: GUIDELINE.md (hunk 1 above)
Comment (islammohamed) [reply GUIDELINE.md]: Great idea, thanks

**Review by BogdanOros — APPROVED** [review]

**Pull request merged**